warehouse/packaging/views.py

```diff
@@ -36,17 +36,33 @@ def project_detail(project, request):
         return HTTPMovedPermanently(request.current_route_path(name=project.name))
 
     try:
+        # Try to find a non-yanked release
         release = (
             request.db.query(Release)
             .filter(Release.project == project, Release.yanked.is_(False))
```

Comment:
If you remove ``Release.yanked.is_(False)`` from here, you can make the first condition of the ``ORDER BY`` be ``Release.yanked``. That should implement this behavior without duplicating the query.